src/algebra/algebra/tower.lean

```diff
@@ -296,10 +296,13 @@ section semiring
 
 variables {R S A}
 variables [comm_semiring R] [semiring S] [add_comm_monoid A]
-variables [algebra R S] [module S A] [module R A] [is_scalar_tower R S A]
 
 namespace submodule
 
+section module
+
+variables [module R S] [module S A] [module R A] [is_scalar_tower R S A]
```

Comment:
After this generalization, is commutativity of `R` used any more?

Reply:
nope! fixed